library/src/main/java/com/datatorrent/lib/io/fs/FileSplitterInput.java

```diff
@@ -375,11 +374,18 @@ public void run()
             lastScannedInfo = null;
             numDiscoveredPerIteration = 0;
             for (String afile : files) {
-              String filePath = new File(afile).getAbsolutePath();
-              LOG.debug("Scan started for input {}", filePath);
-              Map<String, Long> lastModifiedTimesForInputDir;
-              lastModifiedTimesForInputDir = referenceTimes.get(filePath);
-              scan(new Path(afile), null, lastModifiedTimesForInputDir);
+              Path filePath = new Path(afile);
+              LOG.debug("Scan started for input {}", filePath.toString());
+              Map<String, Long> lastModifiedTimesForInputDir = null;
+              if (fs.exists(filePath)) {
+                FileStatus fileStatus = fs.getFileStatus(filePath);
+                if (fileStatus.isDirectory()) {
+                  lastModifiedTimesForInputDir = referenceTimes.get(fileStatus.getPath().toString());
+                } else {
+                  lastModifiedTimesForInputDir = referenceTimes.get(fileStatus.getPath().getParent().toString());
```

Comment:
This is not right, in case user has given input as,
/home/myDir, /home/myDir/file1.txt, the scan of second input i.e. /home/myDir/file1.txt will overwrite the reference times for input /home/myDir.